onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -1250,74 +1282,98 @@ including arg name, arg type (contains both type and shape).)pbdoc")
   py::class_<PyInferenceSession>(m, "InferenceSession", R"pbdoc(This is the main class used to run a model.)pbdoc")
       // In Python3, a Python bytes object will be passed to C++ functions that accept std::string or char*
-      // without any conversion. So this init method can be used for model file path (string)
-      // and model content (bytes)
-      .def(py::init([&env](const PySessionOptions& so, const std::string& arg, bool is_arg_file_name) {
-        auto sess = onnxruntime::make_unique<PyInferenceSession>(env, so, arg, is_arg_file_name);
-        RegisterCustomOpDomainsAndLibraries(sess.get(), so);
-        return sess;
-      }))
-      .def(
-          "load_model", [](PyInferenceSession* sess, std::vector<std::string>& provider_types) {
-            OrtPybindThrowIfError(sess->GetSessionHandle()->Load());
-            InitializeSession(sess->GetSessionHandle(), provider_types);
-          },
-          R"pbdoc(Load a model saved in ONNX format.)pbdoc")
-      .def(
-          "load_model", [](PyInferenceSession* sess, std::vector<std::string>& provider_types, ProviderOptionsVector& provider_options) {
-            OrtPybindThrowIfError(sess->GetSessionHandle()->Load());
-            InitializeSession(sess->GetSessionHandle(), provider_types, provider_options);
-          },
-          R"pbdoc(Load a model saved in ONNX format.)pbdoc")
-      .def("run", [](PyInferenceSession* sess, std::vector<std::string> output_names, std::map<std::string, py::object> pyfeeds, RunOptions* run_options = nullptr) -> std::vector<py::object> {
-        NameMLValMap feeds;
-        for (auto _ : pyfeeds) {
-          OrtValue ml_value;
-          auto px = sess->GetSessionHandle()->GetModelInputs();
-          if (!px.first.IsOK() || !px.second) {
-            throw std::runtime_error("Either failed to get model inputs from the session object or the input def list was null");
-          }
-          CreateGenericMLValue(px.second, GetAllocator(), _.first, _.second, &ml_value);
-          if (PyErr_Occurred()) {
-            PyObject *ptype, *pvalue, *ptraceback;
-            PyErr_Fetch(&ptype, &pvalue, &ptraceback);
-
-            PyObject* pStr = PyObject_Str(ptype);
-            std::string sType = py::reinterpret_borrow<py::str>(pStr);
-            Py_XDECREF(pStr);
-            pStr = PyObject_Str(pvalue);
-            sType += ": ";
-            sType += py::reinterpret_borrow<py::str>(pStr);
-            Py_XDECREF(pStr);
-            throw std::runtime_error(sType);
-          }
-          feeds.insert(std::make_pair(_.first, ml_value));
-        }
+      // without any conversion. So this init method can be used for model file path (string) and model content (bytes)
+      .def(py::init([&env](const PySessionOptions& so, const std::string arg, bool is_arg_file_name,
+                           bool load_config_from_model = false) {
+        std::unique_ptr<PyInferenceSession> sess;
```

Comment:
GitHub diff for this is confusing as well. No change to 'run'. 
init now constructs the InferenceSession, adds any custom op info and loads the model
load_model is replaced by initialize_session. 
This will allow changing the Python interface to enable setting EPs and EP options without recreating the whole InferenceSession instance in the future as that can be done between init and initialize_session calls.